File: demo-gateway-phpclient/vo/BatchVo.php
<?php
namespace vo;

class BatchVo implements \JsonSerializable {
    private $batch_no;
    private $total_count;
    private $request_time;
    private $total_amount;
    private $details;

    /**
     * 需要实现此方法，以便json_encode()方法能返回私有属性
     * @return array|mixed
     */
    public function jsonSerialize(){
        $data = [];
        foreach ($this as $key => $val){
            $data[$key] = $val;
        }
        return $data;
    }

    /**
     * @return mixed
     */
    public function getBatchNo()
    {
        return $this->batch_no;
    }

    /**
     * @param mixed $batch_no
     */
    public function setBatchNo($batch_no)
    {
        $this->batch_no = $batch_no;
    }

    /**
     * @return mixed
     */
    public function getTotalCount()
    {
        return $this->total_count;
    }

    /**
     * @param mixed $total_count
     */
    public function setTotalCount($total_count)
    {
        $this->total_count = $total_count;
    }

    /**
     * @return mixed
     */
    public function getRequestTime()
    {
        return $this->request_time;
    }

    /**
     * @param mixed $request_time
     */
    public function setRequestTime($request_time)
    {
        $this->request_time = $request_time;
    }

    /**
     * @return mixed
     */
    public function getTotalAmount()
    {
        return $this->total_amount;
    }

    /**
     * @param mixed $total_amount
     */
    public function setTotalAmount($total_amount)
    {
        $this->total_amount = $total_amount;
    }

    /**
     * @return mixed
     */
    public function getDetails()
    {
        return $this->details;
    }

    /**
     * @param mixed $details
     */
    public function setDetails(array $details)
    {
        $this->details = $details;
    }


}

File: demo-gateway-backend/src/main/java/org/study/demo/gateway/backend/vo/BatchVo.java
package org.study.demo.gateway.backend.vo;

import java.util.List;

public class BatchVo {
    private String batchNo;
    private Integer totalCount;
    private String requestTime;
    private String totalAmount;
    private List<DetailVo> details;

    public String getBatchNo() {
        return batchNo;
    }

    public void setBatchNo(String batchNo) {
        this.batchNo = batchNo;
    }

    public String getRequestTime() {
        return requestTime;
    }

    public void setRequestTime(String requestTime) {
        this.requestTime = requestTime;
    }

    public Integer getTotalCount() {
        return totalCount;
    }

    public void setTotalCount(Integer totalCount) {
        this.totalCount = totalCount;
    }

    public String getTotalAmount() {
        return totalAmount;
    }

    public void setTotalAmount(String totalAmount) {
        this.totalAmount = totalAmount;
    }

    public List<DetailVo> getDetails() {
        return details;
    }

    public void setDetails(List<DetailVo> details) {
        this.details = details;
    }
}
